### PR TITLE
Add import page for wallets

### PR DESCRIPTION
**Pull request by Zett98**
###  Description
Add import page to have the ability to import pre-existing wallets.

### CODE REVIEW

**Review by rueshyna — COMMENTED** [review]
I leave some comments. others looks good to me. 
There is no critical issues need to be fixed besides the one we discuss offline, new CSS breaking wallet page needs to be fixed.
But it would be nice that the comments can be taken care in the near future iteration if it makes sense to our app. In this case, if the comments are not going to be fixed, please create issues to record. Then, we can move on.

File: components/create/aliases.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                                                        placeholder={validator.address || "Validator address"}
+                                                        default={validator.address}
+                                                    />
+                                                    <ErrorMessage name={`validators.${index}.address`} render={x => {
+                                                        return renderError(x)
+                                                    }} />
+                                                </div>
+                                                <button
+                                                    type="button"
+                                                    className={(errors.validators && errors.validators[index] && get(errors.validators[index]) ? "my-auto" : "") + " bg-primary font-medium text-white p-1.5 md:self-end self-center justify-self-end block md:mx-auto mx-none "}
```
[the remark this answers is not in the document]
Comment (Zett98) [reply components/create/aliases.tsx]: updated

File: components/create/createLoader.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                    <path d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z" fill="currentFill" />
+                </svg>
+                <span className="sr-only">Loading...</span>
+            </div>
+        );
+    }
+    return address.status === 1 ? (
+        <div className="text-sm md:text-xl my-auto text-gray-800 font-bold">{`Wallet successfully created! ${address.address}`}</div>
```
[the remark this answers is not in the document]
Comment (Zett98) [reply components/create/createLoader.tsx]: updated

File: components/import/importLoader.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                    <path d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z" fill="currentFill" />
+                </svg>
+                <span className="sr-only">Loading...</span>
+            </div>
+        );
+    }
+    return address.status === 1 ? (
+        <div className="text-sm md:text-xl my-auto text-gray-800 font-bold">{`Wallet successfully imported! ${address.address}`}</div>
```
[the remark this answers is not in the document]
Comment (Zett98) [reply components/import/importLoader.tsx]: updated

File: components/import/importLoader.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    let state = useContext(AppStateContext);
+    let dispatch = useContext(AppDispatchContext);
+    let [address, setAddress] = useState({ status: 0, address: formState!.walletAddress! });
+    let [loading, setLoading] = useState(true);
+
+    useEffect(() => {
+        (async () => {
+            if (loading && address.status == 0) {
+                try {
+                    console.log('#1')
```
[the remark this answers is not in the document]
Comment (Zett98) [reply components/import/importLoader.tsx:19]: as soon as  issues with networking are solved

File: components/create/aliases.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+                                                    <label className="text-white">Owner Name</label>
+                                                    <Field
+                                                        name={`validators.${index}.name`}
+                                                        className="border-2 p-2 text-sm md:text-md"
+                                                        placeholder={validator.name || "Owner Name"}
+                                                    />
+                                                    <ErrorMessage name={`validators.${index}.name`} render={renderError} />
+                                                </div>
+                                                <div className="relative flex flex-col w-full md:w-auto md:grow justify-start">
+                                                    <label className="text-white" htmlFor={`validators.${index}.address`}>Validator Address</label>
```
[the remark this answers is not in the document]
Comment (Zett98) [reply components/create/aliases.tsx]: updated

File: components/create/aliases.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+                                                    <ErrorMessage name={`validators.${index}.address`} render={x => {
+                                                        return renderError(x)
+                                                    }} />
+                                                </div>
+                                                <button
+                                                    type="button"
+                                                    className={(errors.validators && errors.validators[index] && get(errors.validators[index]) ? "my-auto" : "") + " bg-primary font-medium text-white p-1.5 md:self-end self-center justify-self-end block md:mx-auto mx-none "}
+                                                    onClick={e => {
+                                                        e.preventDefault()
+                                                        remove(index)
```
[the remark this answers is not in the document]
Comment (Zett98) [reply components/create/aliases.tsx:113]: updated

File: components/create/aliases.tsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+                        </FieldArray>
+                    </div>
+                    <div>
+                        <Field component="select" name="requiredSignatures" values={values.requiredSignatures}>
+                            {values.validators.map((_, idx) => <option key={idx + values.validators.length} label={`${idx + 1}/${values.validators.length}`} value={idx + 1}>{idx + 1}/{values.validators.length}</option>)}
+                        </Field>
+                    </div>
+                    <button
+                        className="bg-primary font-medium text-white my-2 p-2 "
+                        type="submit"
```
[the remark this answers is not in the document]
Comment (Zett98) [reply components/create/aliases.tsx]: updated

File: components/import/aliases.tsx [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+                        <FieldArray name="validators">
+                            {() => (
+                                <div className="min-w-full">
+                                    {values.validators.length > 0 && values.validators.map((validator, index) => {
+                                        return (
+                                            <div className=" border-4 border-dashed border-white md:rounded-none md:border-none md:p-none p-2 flex md:flex-row flex-col justify-start items-start min-w-full" key={index}>
+                                                <div className="flex flex-col">
+                                                    <label className="text-white">Owner Name</label>
+                                                    <Field
+                                                        name={`validators.${index}.name`}
```
[the remark this answers is not in the document]
Comment (Zett98) [reply components/import/aliases.tsx:84]: > I think I have a bit lost here.
I created a wallet, the owner alias iscc(me) and address is tz1abc.
Next time, when I create another new wallet, I can see there is a default addresstz1abc with default alias cc
but if I import a wallet and I am one of owners of it, the name of validator is empty?

good catch, i fixed the alias on import

**Pull request merged**